### PR TITLE
jobs: refactor read and write execution detail helpers

### PR DESCRIPTION
**Pull request by adityamaru**
This change moves the logic that breaks down a file's contents into smaller chunks before writing them to the job_info table to a helper method. This is done so that any file written to the job_info table, not just "execution detail" files can use this helper. A similar change is made to the read path. Previously, the read and write methods implicitly wrote to `~profiler/` prefixed info key which meant that the file would show up on the `Advanced Debugging` tab of the job details page in the DBConsole. We don't want all files written to job_info to show up there.

This change also changes the method signature to take in a txn instead of a db so multiple files can be read/written within the scope of a single txn.

Release note: None
Informs: #109671

### CODE REVIEW

**Comment by adityamaru** [comment]
@msbutler this is cleanup I pulled out to address your comment in https://github.com/cockroachdb/cockroach/pull/108863

File: pkg/jobs/job_info_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			name: "file greater than 1MiB",
+			data: make([]byte, 1<<20+1), // 1 MiB + 1 byte
+		},
+	}
+
+	db := s.InternalDB().(isql.DB)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if len(tt.data) > 0 {
+				_, err := rand.Read(tt.data)
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/jobs/job_info_utils_test.go:63]: same nit applies to the execution_detail_utils_test

File: pkg/jobs/job_info_utils.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+) error {
+	jobInfo := InfoStorageForJob(txn, jobID)
+
+	var chunkCounter int
+	var chunkName string
+	for len(data) > 0 {
+		chunkSize := bundleChunkSize
+		chunk := data
+		if len(chunk) > chunkSize {
+			chunkName = fmt.Sprintf("%s#%04d", filename, chunkCounter)
```
Comment (msbutler) [inline pkg/jobs/job_info_utils.go:42]: so we cant have more than 999 chunks, given the chunk name. Should this function through an error if we reach that state?

**Review by msbutler — COMMENTED** [review]
looks great! left minor comments

File: pkg/jobs/job_info_utils_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		{
+			name: "file less than 1MiB",
+			data: make([]byte, 1<<20-1), // 1 MiB - 1 byte
+		},
+		{
+			name: "file equal to 1MiB",
+			data: make([]byte, 1<<20), // 1 MiB
+		},
+		{
+			name: "file greater than 1MiB",
```
[the remark this answers is not in the document]
Comment (adityamaru) [reply pkg/jobs/job_info_utils_test.go:54]: Test runs in 2 seconds so I'm not too concerned

File: pkg/jobs/job_info_utils_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (adityamaru) [reply pkg/jobs/job_info_utils_test.go:63]: done

File: pkg/jobs/job_info_utils.go (hunk 2 above)
Comment (adityamaru) [reply pkg/jobs/job_info_utils.go:42]: good point, but I think its 9999 chunks, which is ~10gigs with each chunk being 1MB. Added an assertion failure for if we ever hit this limit.

File: pkg/jobs/job_info_utils_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadWriteChunkedFileToJobInfo(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	rng, _ := randutil.NewTestRand()
```
Comment (msbutler) [inline pkg/jobs/job_info_utils_test.go:31]: nit: i beleive you need to log the seed, so that upon test failure inspection, you can easily repro

**Review by msbutler — APPROVED** [review]

File: pkg/jobs/job_info_utils_test.go (hunk 4 above)
Comment (msbutler) [reply pkg/jobs/job_info_utils_test.go:31]: oh dang! good find. you're right.

**Review by msbutler — APPROVED** [review]

**Comment by adityamaru** [comment]
TFTR!

bors r=msbutler

**Pull request merged**